### DISCUSSION TITLE
Cloud compute

### DISCUSSION THREAD

**Discussion by viko42** [Snowad14/IAStreameur#1] [Ideas]
# PR en vue : Faire tourner wav2lip et générer de la voix avec vast.ai, ça vous dit ?

Salut tout le monde,

J'ai une idée pour une PR. Vous savez, j'utilise vast.ai pour louer des cartes graphiques à 0,40 $ de l'heure, et je me disais que ce serait cool d'ajouter ça dans notre projet pour les calculs wav2lip et la génération de voix.

## Les + :

1. **Pas cher**: On peut avoir accès à des CG sans se ruiner.
2. **Plus de monde peut participer**: Pas besoin d'un super PC pour contribuer au projet.
3. **On choisit ce qu'on veut**: Sur vast.ai, tu choisis la carte qui te plaît.
4. **Simple à mettre en place**: Ils ont une API et un CLI qui sont faciles à utiliser.

## Les - :

1. **On dépend d'un autre service**: Si vast.ai change ses tarifs ou ferme
2. **Ça complique un peu les choses**: Plus de code à gérer

**Comment by Snowad14** [comment]
C'est ce que j'ai aussi proposé de faire à Defend sur son chat. Mais le truc, c'est que pour l'instant, le GPU n'a vraiment pas besoin d'être performant, car la mise en cache des données de détection des visages rend le processus 10 fois plus rapide. Je compte rajouter VITS+RVC, qui est un TTS assez peu coûteux en termes de ressources, donc en local ça devrait aller. Mais si quelqu'un veut vraiment le faire, il peut simplement cloner le dépôt et faire exactement la même chose en local, mais sur vast.ai, et se connecter avec son navigateur au serveur web distant. 
Ce qui peut être intéressant, en revanche, c'est de créer une image Docker pour que l'installation se fasse en un clic sur vast.ai
